### Import Pandas and Load Dataset

In [39]:
import pandas as pd

In [51]:
df = pd.read_csv('fao.csv',encoding='latin-1')

# inspect the top few rows of the dataframe
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

### Q11. What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [61]:
df.groupby('Item')[['Y2014','Y2017']].sum().loc['Animal fats']

Y2014    209460.54
Y2017    269617.53
Name: Animal fats, dtype: float64

Answer: **209460.54** and **269617.53**

### Q12. What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [42]:
df.describe().Y2015[['mean','std']].round(3)

mean     135.236
std     1603.404
Name: Y2015, dtype: float64

Answer: **135.236** and **1603.404**

### Q13. What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [43]:
total_observations = df.shape[0]
missing_2016 = df.isna().sum().Y2016
percent_missing = (missing_2016*100/total_observations).round(2)
missing_2016,percent_missing

(1535, 2.52)

Answer: **1535** and **2.52**

### Q14. Which year had the highest correlation with ‘Element Code’?

In [44]:
abs(df.corr()['Element Code']).sort_values(ascending=False)

Element Code    1.000000
Item Code       0.024683
Y2014           0.024457
Y2018           0.024279
Y2017           0.024254
Y2015           0.023889
Y2016           0.023444
Area Code       0.000209
Name: Element Code, dtype: float64

Answer: **2014**

### Q15. What year has the highest sum of Import Quantity??

In [45]:
df.groupby('Element').sum().loc['Import Quantity','Y2014':'Y2018'].T.sort_values(ascending=False)

Y2017    294559.09
Y2018    287997.09
Y2016    286582.78
Y2014    274144.48
Y2015    267018.46
Name: Import Quantity, dtype: float64

Answer: **2017**

### Q16. What is the total number of the sum of Production in 2014?

In [46]:
df.groupby('Element').sum().loc['Production','Y2014']

1931287.75

Answer: **1931287.75**

### Q17. Which of these elements had the highest sum in 2018?

In [47]:
df[['Y2018','Element']].groupby('Element').sum().sort_values(by='Y2018',ascending=False)

Y2018
Element                                           
Domestic supply quantity                2161192.10
Production                              2075072.89
Food                                    1303841.28
Total Population - Both sexes           1140605.00
Food supply (kcal/capita/day)            455261.00
Processing                               308429.00
Import Quantity                          287997.09
Feed                                     233489.68
Export Quantity                          181594.80
Losses                                   163902.00
Other uses (non-food)                     91300.97
Food supply quantity (kg/capita/yr)       49056.85
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Protein supply quantity (g/capita/day)    11833.56
Fat supply quantity (g/capita/day)        10258.69
Tourist consumption                          90.00

Answer: **Domestic supply quantity**

### Q18. Which of these elements had the 3rd lowest sum in 2018?

In [48]:
df[['Y2018','Element']].groupby('Element').sum().sort_values(by='Y2018')

Y2018
Element                                           
Tourist consumption                          90.00
Fat supply quantity (g/capita/day)        10258.69
Protein supply quantity (g/capita/day)    11833.56
Stock Variation                           20577.91
Seed                                      25263.14
Residuals                                 34864.00
Food supply quantity (kg/capita/yr)       49056.85
Other uses (non-food)                     91300.97
Losses                                   163902.00
Export Quantity                          181594.80
Feed                                     233489.68
Import Quantity                          287997.09
Processing                               308429.00
Food supply (kcal/capita/day)            455261.00
Total Population - Both sexes           1140605.00
Food                                    1303841.28
Production                              2075072.89
Domestic supply quantity                2161192.10

Answer: **Protein supply quantity (g/capita/day)**

### Q19. What is the total Import Quantity in Algeria in 2018?

In [49]:
df[df.Area=='Algeria'].groupby('Element').Y2018.sum()['Import Quantity']

36238.29

Answer: **36238.29**

### Q20. What is the total number of unique countries in the dataset?

In [50]:
df.Area.nunique()

49

Answer: **49**